### DISCUSSION TITLE
What is the best hosting option for a grapevine console app?

### DISCUSSION THREAD

**Discussion by ertan2002** [scottoffen/grapevine#101] [Q&A] [answered]
Hi all

I've created a small rest app by using grapevine (it is an amazing library, thank you a lot)..

I am not sure what is the best approach. I am thinking to convert my console app to windows service app (I think, it should work).. I do not trust a regular console app because if somebody press enter mistakenly, the app will be terminated.. 

What kind of hosting are you using for windows server? 

thanks

**Comment by seikosantana** [comment]
I don't think relying on Console.ReadLine() (if you do) for holding the process to keep running is a good idea. I've been using Grapevine for like 5 projects and I will use IRestServer.Run() instead to keep the server running.

I host applications on Ubuntu and not on Windows. I use UpCloud for the hosting (I see that it has Windows Server too).
On Ubuntu, I will be registering the console application as a systemd service. Last time I host my Grapevine application on Windows Server, I was hosting it on a Ubuntu VM over Hyper-V. While this might not be an answer to your specific question about Windows Server hosting, I hope this is useful.

**Comment by scottoffen** [comment, marked as answer]
I'd recommend using a [Worker Service](https://docs.microsoft.com/en-us/dotnet/core/extensions/workers). They have the advantage of being able to be run on both Windows and Linux. I have a sample project I'm working on for it, but it is incomplete. Nevertheless, using the documentation linked to above, it should be pretty easy to work out how to do it.
